src/list.ts

```diff
@@ -0,0 +1,259 @@
+            data.anzahl = Number.parseInt(quantity.value);
+            data.artikelname = productName.value;
+            await updateListEntry(data);
+        }
+        productName.disabled = !productName.disabled.valueOf();
+        quantity.disabled = !quantity.disabled.valueOf();
+    });
+
+    addToRow(checkbox, row);
+    addToRow(productName, row);
+    addToRow(quantity, row);
+    addToRow(editButton, row);
+    addToRow(deleteButton, row);
+
+    return row;
+}
+
+function addToRow(element: HTMLElement, row: HTMLTableRowElement) {
+    const td = document.createElement('td');
+    td.appendChild(element);
+    row.appendChild(td);
+}
+
+function addRowToTable(newEntry: Listeneintrag, id?: string) {
+    if (id == undefined) {
+        productTable.appendChild(createTableRow(newEntry));
+        return;
+    }
+    productTable.appendChild(createTableRow(newEntry, id));
+}
+
+function realtimeUpdate(list: Einkaufsliste) {
+    const listChannel = 'databases.' + CONFIG.DATABASE_ID +
+        '.collections.' + CONFIG.DB_COLLECTION_SHOPPINGLISTS +
+        '.documents.' + list.$id!;
+    const documentChannel = 'databases.' + CONFIG.DATABASE_ID +
+        '.collections.' + CONFIG.DB_COLLECTION_SHOPLISTENTRY +
+        '.documents';
+    client.subscribe([listChannel, documentChannel], response => {
+        if (response.channels.includes(listChannel)) {
+            const changedList = response.payload as Einkaufsliste;
+            updateListInfo(changedList);
+        }
+        if (response.channels.includes(documentChannel)) {
+            documentEventHandler(response, list.$id!);
+        }
+    });
+}
+
+function documentEventHandler(response: RealtimeResponseEvent<unknown>, id: string) {
+    const changedEntry = response.payload as Listeneintrag;
+    if (changedEntry.$id != id) {
+        return;
+    }
+    switch (true) {
+        case /create$/.test(response.events[0]):
+            addRowToTable(changedEntry);
+            console.log('It creates!');
+            break;
+        case /delete$/.test(response.events[0]):
+            deleteRow(changedEntry.$id!);
+            console.log('It deletes!');
+            break;
+        case /update$/.test(response.events[0]):
+            updateRowEntry(changedEntry);
+            console.log('It updates!');
+            break;
+        default:
+            break;
+    }
+}
+
+function deleteRow(id: string) {
+    const deletedElement = document.getElementById(id) as HTMLTableRowElement;
+    productTable.removeChild(deletedElement);
+}
+
+function updateRowEntry(entry: Listeneintrag) {
+    const tableRow = document.getElementById(entry.$id!);
+    const checkbox = tableRow?.children[0].firstChild as HTMLInputElement;
+    checkbox.checked = entry.erledigt;
+
+    const name = tableRow?.children[1].firstChild as HTMLInputElement;
+    name.value = entry.artikelname;
+
+    const quantity = tableRow?.children[2].firstChild as HTMLInputElement;
```

Comment:
I don't like how you're addressing the children by their index. This code would break far too easily, if anything changed about the row html.

A querySelector might be more robust.